assets/js/admin-autoshare-for-twitter-classic-editor.js

```diff
@@ -125,15 +125,41 @@
 	}
 
 	/**
-	 * Updates the counter
+	 * Calculates the permalink length
 	 */
-	function updateRemainingField() {
+	function getPermalinkLength() {
 		let permalinkLength = 0;
 		if ( $('#sample-permalink').length ) {
-			permalinkLength = $('#sample-permalink').text().length
+			if (adminAutoshareForTwitter.twitterURLLength) {
```

Comment:
As all URLs are wrapped in t.co links. I think we can remove the fallback logic here. However, I noticed that Twitter doesn't wrap inaccessible URLs (local wp setup), for those cases we may still want to go with the current length counter logic.

So, the best way here is to keep our current url chars counter for local setups and Fix 23 chars in live sites. what do you think?

Reply:
We can add a function given below in Utils.php to identify the wp environment type.
```php
/**
 * Returns whether WP environment is local.
 *
 * @return boolean
 */
function is_local() {
	$is_local_env = in_array( wp_get_environment_type(), [ 'local', 'development' ], true );
	$is_local_url = strpos( home_url(), '.test' ) || strpos( home_url(), '.local' );
	$is_local     = $is_local_env || $is_local_url;

	/**
	 * Filters whether WP environment is local or not.
	 *
	 * @param bool   whether WP environment is local or not.
	 */
	return apply_filters( 'autoshare_for_twitter_is_local', $is_local );
}

```

Reply:
Ah, interesting. Yes, that does make sense.